qhana_plugin_registry/tasks/plugin_filter.py

```diff
@@ -44,10 +44,14 @@ def get_plugins_from_filter(filter_dict: dict, plugin_mapping: dict) -> set[RAMP
 
     match filter_dict:
         case {"and": filter_expr}:
+            if not filter_expr:
```

Comment:
You may want to write unit tests for the plugin filter functions to test the edge cases properly.